RFC/src/RFC-0173_Versioning.md

```diff
@@ -0,0 +1,94 @@
+
+### WireMode byte
+In the Bitcoin P2P protocol messages are preceded by 4 [magic values](https://en.bitcoin.it/wiki/Protocol_documentation#Common_structures) or bytes. These values are used to 
+delimit when a new message starts in a byte stream and also are used to indicate which of the Bitcoin networks the node is speaking on, such as TestNet or Mainnet.
+
+Tari message packets are encapsulated using the Noise protocol so we do not need the delimiting functionality of these bytes but Tari will include a single WireMode byte at the beginning
+of every connection session. This byte will indicate which network a node is communicating on, so that if the counterparty is on a different network it can reject this connection cheaply without
+having to perform any further operations, like completing the Noise protocol handshake.
+
+### P2P message version
+Peer to Peer messages on the Tari network are encapsulated into message envelopes. The body of message envelopes are defined, serialized and deserialized using Protobuf. These messages will only be updated by adding 
+new fields to the Protobuf definitions, never removing fields. This is done in order to preserve backwards compatibility where newer nodes can still communicate with older nodes, but the reverse
+is not true. As the P2P messaging protocol evolves fields might fall out of use. In order to keep track of which fields are currently being used, each P2P message header contains a `version` field. 
+As major changes are made to the messaging formats this version will increase to allow clients to know if they are able to properly deserialize these messages.
+
+### Consensus version
+The final aspect of the Tari system that will be versioned are the Consensus rules. These rules will change as the network matures. Changes to consensus rules can be achieved using either a Soft fork or Hard fork. 
+Soft forks are where new consensus rules are added that older nodes will see as valid (thus backwards compatible) but newer nodes will reject blocks from older nodes that are not aware of the new consensus rules. 
+A hard fork means that the new consensus rules are not backwards compatible and so only nodes that know about the new rules will be able to produce and validate new transactions and blocks.
+
+The consensus version will be used by a node to determine if it can interact with another node successfully or not. A list of fork versions will be maintained within the code. When a connection is 
+started with a new node the two nodes will exchange `Version` messages detailing the consensus version they are each running. Both nodes will need to reply with a `Version Acknowledge` message to confirm that they are
+compatible with the counterparty's version. Only once the acknowledgments have been exchanged can further messages be exchanged by the parties. This is the method currently employed on 
+the [Bitcoin network](https://developer.bitcoin.org/devguide/p2p_network.html#connecting-to-peers)
```

Comment:
Here it makes sense to inc version numbers by full integers, and not maintain a minor version.

I suggest you add a paragraph that makes the handling explicit,

Node A will talk to Node B  IFF A recognizes the version AND B.blockheight is in the range where that version was active, according to A's fork version list. This means that B shouldn't reject A immediately; it should wait for A to reject him (since A might be able to downgrade) before killing the connection.

Reply:
Agree

Reply:
Added a paragraph with explicit example and explicitly stated a single incrementing integer